Mathlib/FieldTheory/IsSepClosed.lean

```diff
@@ -191,6 +199,13 @@ class IsSepClosure [Algebra k K] : Prop where
 instance IsSepClosure.self_of_isSepClosed [IsSepClosed k] : IsSepClosure k k :=
   ⟨by assumption, isSeparable_self k⟩
 
+-- TODO:
+/- If `k` is perfect, `K` is a separable closure of `k`,
+then it is also an algebraic closure of `k`. -/
+-- instance IsSepClosure.isAlgClosure_of_perfectField [IsSepClosure k K] [PerfectField k] :
+--     IsAlgClosure k K :=
+--   sorry
+
```

Comment:
maybe use `proof_wanted`?

Reply:
But it complains about every variable being unused. Will it trigger "warning-as-error" error?

Reply:
It turns out that this result cannot be proved inside this file. 😅 Separable degree argument is required.

Reply:
Why? I think combining [IsAlgClosed.of_exists_root](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/IsAlgClosed/Basic.html#IsAlgClosed.of_exists_root) with the definition [PerfectField](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/Perfect.html#PerfectField) gives a proof.

Reply:
> Why? I think combining [IsAlgClosed.of_exists_root](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/IsAlgClosed/Basic.html#IsAlgClosed.of_exists_root) with the definition [PerfectField](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/Perfect.html#PerfectField) gives a proof.

No, these two results in their current form can't solve this problem. Two ways to fix it:
- Prove that if `K / k` is algebraic, `k` is perfect, then `K` is also perfect. This seems to require separable degree argument.
- Prove a modified version of `IsAlgClosed.of_exists_root` which states that if `K / k` is algebraic, any monic irreducible polynomial over `k` has a root in `K`, then `K` is algebraically closed (in fact an algebraic closure of `k`).

I think both of them are important results, but not in mathlib yet.

Reply:
Oh I see, I think I read `PerfectField k` as `PerfectField K`. For your first option, you just need to know that every irreducible polynomial in K[X] divides some irreducible polynomial in k[X]. You can do this by adjoining a root of the first polynomial to K and take its minpoly over k.

Reply:
The stronger version of your second option is also true, see Theorem 2 in https://kconrad.math.uconn.edu/blurbs/galoistheory/algclosure.pdf (would be nice to have it in mathlib!)

Reply:
> Oh I see, I think I read `PerfectField k` as `PerfectField K`. For your first option, you just need to know that every irreducible polynomial in K[X] divides some irreducible polynomial in k[X]. You can do this by adjoining a root of the first polynomial to K and take its minpoly over k (we know it's integral over k by [isIntegral_trans](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/IntegralClosure.html#isIntegral_trans)).

Thank you very much, it works!



> The stronger version of your second option is also true, see Theorem 2 in https://kconrad.math.uconn.edu/blurbs/galoistheory/algclosure.pdf, but it takes one page and a half to prove; would be nice to have it in mathlib!

Let me add a TODO to `IsAlgClosed`. I think there should also be a `IsSepClosed` version?

Reply:
It seems like you can do `set_option linter.unusedVariables false in` (as per Yaël on zulip).